validator/rpc/wallet_test.go

```diff
@@ -83,34 +83,62 @@ func TestServer_CreateWallet_Imported(t *testing.T) {
 	require.NoError(t, err)
 }
 
-func TestServer_CreateWallet_Derived(t *testing.T) {
+func TestServer_RecoverWallet_Derived(t *testing.T) {
 	localWalletDir := setupWalletDir(t)
 	defaultWalletPath = localWalletDir
 	ctx := context.Background()
 	s := &Server{
 		walletInitializedFeed: new(event.Feed),
 		walletDir:             localWalletDir,
 	}
-	req := &pb.CreateWalletRequest{
-		Keymanager:     pb.KeymanagerKind_DERIVED,
+	req := &pb.RecoverWalletRequest{
 		WalletPassword: strongPass,
 		NumAccounts:    0,
 	}
-	// We delete the directory at defaultWalletPath as CreateWallet will return an error if it tries to create a wallet
+	// We delete the directory at defaultWalletPath as RecoverWallet will return an error if it tries to create a wallet
 	// where a directory already exists
 	require.NoError(t, os.RemoveAll(defaultWalletPath))
-	_, err := s.CreateWallet(ctx, req)
+	_, err := s.RecoverWallet(ctx, req)
 	require.ErrorContains(t, "Must create at least 1 validator account", err)
 
 	req.NumAccounts = 2
-	_, err = s.CreateWallet(ctx, req)
-	require.ErrorContains(t, "Must include mnemonic", err)
+	req.Language = "Swahili"
+	_, err = s.RecoverWallet(ctx, req)
+	require.ErrorContains(t, "input not in the list of supported languages", err)
+
+	req.Language = "ENglish"
+	_, err = s.RecoverWallet(ctx, req)
+	require.ErrorContains(t, "invalid mnemonic in request", err)
 
 	mnemonicResp, err := s.GenerateMnemonic(ctx, &empty.Empty{})
 	require.NoError(t, err)
 	req.Mnemonic = mnemonicResp.Mnemonic
 
-	_, err = s.CreateWallet(ctx, req)
+	req.Mnemonic25ThWord = " "
+	req.SkipMnemonic25ThWord = false
+	_, err = s.RecoverWallet(ctx, req)
+	require.ErrorContains(t, "mnemonic 25th word cannot be empty", err)
+	req.Mnemonic25ThWord = "outer"
+
+	//test weak password
+	req.WalletPassword = "123qwe"
+	_, err = s.RecoverWallet(ctx, req)
+	require.ErrorContains(t, "password did not pass validation", err)
+
+	req.WalletPassword = strongPass
+	//create(drived should fail) then test recover
```

Comment:
Let's try to use consistent formatting for comments, with capitalization and punctuation with spacing as needed